src/main/java/seedu/address/logic/commands/room/SearchRoomCommand.java

```diff
@@ -0,0 +1,61 @@
+package seedu.address.logic.commands.room;
+
+import static java.util.Objects.requireNonNull;
+import static seedu.address.commons.core.Messages.MESSAGE_INVALID_ROOM_NOT_FOUND;
+
+import java.util.List;
+
+import seedu.address.commons.core.index.Index;
+import seedu.address.logic.commands.Command;
+import seedu.address.logic.commands.CommandResult;
+import seedu.address.logic.commands.exceptions.CommandException;
+import seedu.address.model.Model;
+import seedu.address.model.room.Room;
+
+
+/**
+ * Searches a room according to the given room number.
+ */
+public class SearchRoomCommand extends Command {
+
+    public static final String COMMAND_WORD = "searchroom";
+
+    public static final String MESSAGE_USAGE = COMMAND_WORD
+            + ": Finds the room with the given room number.\n"
+            + "Parameters: ROOM NUMBER\n"
+            + "Example: " + COMMAND_WORD + " 130";
+
+    public static final String MESSAGE_SUCCESS = "Room %1$s has been found and listed.";
+
+    private final Integer roomNumber;
+
+    /**
+     * Creates a SearchRoomCommand to look for the specified room.
+     * @param roomNumber the room to find.
+     */
+    public SearchRoomCommand(Integer roomNumber) {
+        requireNonNull(roomNumber);
+        this.roomNumber = roomNumber;
+    }
+
+    //TODO
+    @Override
+    public CommandResult execute(Model model) throws CommandException {
+        Index index = model.getModifiableRoomList().checkIfRoomPresent(roomNumber);
+        List<Room> lastShownList = model.getModifiableRoomList().getRoomObservableList();
```

Comment:
maybe you can use `model.getUnmodifiableRoomList()` here? `model.getUnmodifidableRoomList()` returns an ObservableList<Room> too. i discussed with yh last time about how the `command` class shouldnt be interacting directly with the room class but rather through the `model` class

Reply:
Yep yep the search room command is incomplete as I've mentioned haha. The main thing in this PR is just refactoring.

Reply:
ah, i see i see, sorry!!